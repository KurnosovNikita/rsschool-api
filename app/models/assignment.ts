import { Document, Schema, model } from 'mongoose';

export enum AssignmentStatus {
    Assigned = 'Assigned',
    ReadyForReview = 'ReadyForReview',
    Rejected = 'Rejected',
    Checked = 'Checked',
}

export interface IAssignment {
    assignmentRepo?: string;
    checkDate?: number;
    completeDate?: number;
    courseId: string;
    deadlineDate: number;
    mentorComment?: string;
    mentorId?: string;
    score: number;
    status: AssignmentStatus;
    studentComment?: string;
    studentId: string;
    taskId: string;
}

export interface IAssignmentModel extends Document, IAssignment {
    _id: string;
}

export const AssignmentSchema: Schema = new Schema({
    assignmentRepo: String,
    checkDate: Number,
    completeDate: Number,
    courseId: String,
    deadlineDate: Number,
    mentorComment: String,
    mentorId: String,
    score: Number,
    status: String,
    studentComment: String,
    studentId: String,
    taskId: String,
});

export const AssignmentModelName = 'Assignment';
export const AssignmentModel = model<IAssignmentModel>(AssignmentModelName, AssignmentSchema);
