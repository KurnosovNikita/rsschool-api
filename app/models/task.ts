import { Document, Schema, model } from 'mongoose';

enum TaskType {
    Task = 'Task',
    CodeJam = 'Code Jam',
    Interview = 'Interview',
    Test = 'Test',
}

enum WhoChecks {
    Mentor = 'Mentor',
    RandomMentor = 'Random Mentor',
    Trainer = 'Trainer',
    UnitTest = 'Unit Test',
    ExternalPerson = 'External Person',
    WithoutChecking = 'Without Checking',
    Codewars = 'Codewars',
    Codecademy = 'Codecademy',
    Duolingo = 'Duolingo',
}

export interface ITask {
    author: string;
    courseId: string;
    endDateTime: number;
    mentorCommentTemplate?: string;
    startDateTime: number;
    studentCommentTemplate?: string;
    taskType: TaskType;
    title: string;
    type: string;
    urlToDescription: string;
    whoChecks: WhoChecks;
}

export interface ITaskModel extends Document, ITask {
    _id: string;
}

export const TaskSchema: Schema = new Schema({
    author: String,
    courseId: String,
    endDateTime: Number,
    mentorCommentTemplate: String,
    startDateTime: Number,
    studentCommentTemplate: String,
    taskType: String,
    title: String,
    type: String,
    urlToDescription: String,
    whoChecks: String,
});

export const TaskModelName = 'Task';
export const TaskModel = model<ITaskModel>(TaskModelName, TaskSchema);
