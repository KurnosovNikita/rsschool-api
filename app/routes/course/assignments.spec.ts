import { IRouterContext } from 'koa-router';
import { AssignmentStatus } from '../../models';
import { OK } from 'http-status-codes';
import { setResponse } from '../utils';

const getMockAssignments = jest.fn(() => {
    return [
        {
            _id: 'string',
            assignmentRepo: 'string',
            checkDate: 123,
            completeDate: 123,
            courseId: 'string',
            deadlineDate: 123,
            mentorComment: 'string',
            mentorId: 'string',
            score: 123,
            status: AssignmentStatus.Assigned,
            studentComment: 'string',
            studentId: 'string',
            taskId: 'string',
        },
    ];
});

describe('app/routes/course/assignments.ts', () => {
    it('returns data', async () => {
        const ctx = createTestContext();
        const assignments = await getMockAssignments();
        const arrayOfTasksId: string[] = [];
        for (const index in assignments) {
            if (assignments[index]) {
                arrayOfTasksId.push(assignments[index].taskId);
            }
        }
        expect(getMockAssignments).toHaveBeenCalledTimes(1);
        setResponse(ctx, OK, ctx.body);
        expect(ctx.body).toMatchSnapshot();
        expect(ctx.status).toBe(200);
    });
});

export function createTestContext(): IRouterContext {
    return {
        body: {
            assignmentRepo: 'repo',
            checkDate: 123,
            completeDate: 123,
            courseId: 'String',
            deadlineDate: 123,
            mentorComment: 'String',
            mentorId: 'String',
            score: 0,
            status: 'Assigned',
            studentComment: 'String',
            studentId: 'String',
            taskId: 'String',
        },
    } as any;
}
