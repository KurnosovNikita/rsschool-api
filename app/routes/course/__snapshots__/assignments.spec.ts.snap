// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/routes/course/assignments.ts returns data 1`] = `
Object {
  "data": Object {
    "assignmentRepo": "repo",
    "checkDate": 123,
    "completeDate": 123,
    "courseId": "String",
    "deadlineDate": 123,
    "mentorComment": "String",
    "mentorId": "String",
    "score": 0,
    "status": "Assigned",
    "studentComment": "String",
    "studentId": "String",
    "taskId": "String",
  },
}
`;
